mailbox.go

```diff
@@ -43,7 +43,8 @@ type MailboxInfo struct {
 	// The mailbox attributes.
 	Attributes []string
 	// The server's path separator.
-	Delimiter string
+	Delimiter       string
+	HasNILDelimiter bool
```

Comment:
Instead of adding this new field, is it possible to treat an empty `Delimiter` as NIL? Or would that conflict with something else?

Reply:
Not too sure. I didn't want someone who explicitly wants their delimiter to be `""` (for whatever insane reason) to be surprised when it comes out as `NIL`. I don't think the spec says anything about that.

Reply:
Alright, I think there's no need to worry about a `""` delimiter. This means we can avoid introducing the boolean which is much cleaner imo.

Reply:
IMAP says the hierarchy delimiter is a "character", so I think it'd be fine. I'd prefer this to avoid having two sources of truth (what happens if `Delimiter` is non-empty and `HasNILDelimiter` is true?).